core/domain/rights_manager.py

```diff
@@ -207,19 +206,21 @@ def _update_activity_summary(activity_type, activity_rights):
 
 def update_activity_first_published_msec(
         activity_type, activity_id, first_published_msec):
-    """Updates the first_published_msec field for an activity. Callers are
-    responsible for ensuring that this value is not already set before updating
-    it.
+    """Updates the first_published_msec field for an activity. This is only
+    called during the one-off job ExplorationFirstPublishedOneOffJob. Callers
+    are responsible for ensuring that this value is not already set before
+    updating it.
     """
     activity_rights = _get_activity_rights(activity_type, activity_id)
     activity_rights.first_published_msec = first_published_msec
     commit_cmds = [{
         'cmd': CMD_UPDATE_FIRST_PUBLISHED_MSEC,
-        'first_published_msec': first_published_msec
+        'old_first_published_msec': None,
```

Comment:
technically I would suggest using activity_rights.first_published_msec here and moving line 215 after this one